Tools/autotest/common.py

```diff
@@ -2269,8 +2307,81 @@ def test_arm_feature(self):
+                    if mode in self.NOT_DISARMED_SETTABLE_MODES_LIST:
+                        self.progress("Not settable mode : %s" % mode)
+                        try:
+                            self.do_set_mode_via_command_long(mode, timeout=2)
+                        except AutoTestTimeoutException:
+                            self.progress("PASS not arm mode : %s" % mode)
+                            pass
+                        except ValueError:
+                            self.progress("PASS not arm mode : %s" % mode)
+                            pass
+                    else:
+                        self.progress("Not armable mode : %s" % mode)
+                        self.do_set_mode_via_command_long(mode)
+                        self.run_cmd(mavutil.mavlink.MAV_CMD_COMPONENT_ARM_DISARM,
+                                     1,  # ARM
+                                     0,
+                                     0,
+                                     0,
+                                     0,
+                                     0,
+                                     0,
+                                     want_result=mavutil.mavlink.MAV_RESULT_FAILED
+                                     )
+                    self.progress("PASS not arm mode : %s" % mode)
+                if mode in self.POSITION_ARMABLE_MODES_LIST:
+                    self.progress("Armable mode needing Position : %s" % mode)
+                    self.wait_ekf_happy()
+                    self.do_set_mode_via_command_long(mode)
+                    self.arm_vehicle(timeout=30)
+                    if not self.disarm_vehicle():
+                        self.progress("Failed to DISARM")
+                        raise NotAchievedException()
+                    self.progress("PASS arm mode : %s" % mode)
+                    self.progress("Not armable mode without Position : %s" % mode)
+                    self.wait_gps_disable()
+                    self.do_set_mode_via_command_long(mode)
+                    self.run_cmd(mavutil.mavlink.MAV_CMD_COMPONENT_ARM_DISARM,
+                                 1,  # ARM
+                                 0,
+                                 0,
+                                 0,
+                                 0,
+                                 0,
+                                 0,
+                                 want_result=mavutil.mavlink.MAV_RESULT_FAILED
+                                 )
+                    self.set_parameter("SIM_GPS_DISABLE", 0)
+                    self.progress("PASS not arm mode without Position: %s" % mode)
+                if mode == "FOLLOW":
+                    self.set_parameter("FOLL_ENABLE", 0)
+        finally:
+            self.do_set_mode_via_command_long(self.default_mode())
+            if self.armed():
+                if not self.disarm_vehicle():
+                    self.progress("Failed to DISARM")
+                    raise NotAchievedException()
```

Comment:
Might this cloud the other exception that you might currently be handling?

Also, progress should go into exception call.

Reply:
It isn't called when exceptions are raised on not settable mode on disarmed state, so that is fine